parsers/IL.py

```diff
@@ -66,24 +42,45 @@ def fetch_price(
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> dict:
-    """Fetch price from IEC table."""
+    """Fetch price from Noga API."""
     if target_datetime is not None:
         raise NotImplementedError("This parser is not yet able to parse past dates")
 
     with get(IEC_PRICE) as response:
-        soup = BeautifulSoup(response.content, "lxml")
+        data = response.json()
 
-    price = soup.find("td", class_="ms-rteTableEvenCol-6")
+    components = data.get("components")
+    if not components:
+        raise ValueError("Could not parse IEC price")
+    general_message = components[0].get("text")
+    general_message = base64.b64decode(general_message).decode("utf-8")
+
+    date = extract_date_from_string(general_message)
+
+    table = components[1].get('table', [])
+    kw_prices = table[-1]
+    price_with_VAT = kw_prices[-1].get("value")
+    price_with_VAT = base64.b64decode(price_with_VAT).decode("utf-8")
```

Comment:
We should check that kw_prices are not None or N/A before passing it to the decode function. 

I would also avoid defining the price_with_VAT and re-assigning it directly after, this can be inlined.